### DISCUSSION TITLE
Binance: Too many subscriptions

### DISCUSSION THREAD

**Discussion by CloudGiantLabs** [altangent/ccxws#282] [Q&A] [answered]
Hello, I am attempting to subscribe to all of Binance's markets (1396 in total) in a for loop using the following:
`binance.subscribeCandles(market);`
If I try to connect to more than 1024 markets, I get this error:
`msg: { error: { code: 4, msg: 'Too many subscriptions' }, id: 6 }` 

I have even tried subscribing to 200 markets at a time with a delay of 1 minute but the limit seems to be 1025 markets.

Obviously I am hitting the websocket limit Binance has documented here:
https://github.com/binance-us/binance-official-api-docs/blob/master/web-socket-streams.md

But I can't figure out a way around it other than 2 separate servers. Any help would be appreciated.

**Comment by bmancini55** [comment, marked as answer]
Looking at the doco it states

> A single connection can listen to a maximum of 1024 streams.

It looks like that's not limited by IP though.  Can you just create a second client instance?

```javascript
const client1 = new Binance(); // subscribes to first 1024 streams
const client2 = new Binance(); // subscribes to the remaining streams
```

**Reply by nileio** [reply]
that does not work anymore ! I think a new websocket client does not bypass the limit. It is unclear from binance docs what a "single connection" actually mean. it should typically mean a websocket but it is rather an IP from what it looks like
